api/src/main/java/io/jsonwebtoken/JwtParser.java

```diff
@@ -41,7 +41,9 @@
      * @return the parser method for chaining.
      * @see MissingClaimException
      * @see IncorrectClaimException
+     * @deprecated see {@link JwtParserBuilder#requireId(String)}
```

Comment:
I think we should make these deprecation notices super clear - why they're deprecated, and how they should be using the builder instead.  A link to `Jwts#parserBuilder()` and some minor explanation should be ok.

Also, when I add a deprecation notice, I like to add it in the JavaDoc paragraph as well, with very clear (bold) indications that this method will be removed from 1.0 final.